src/analysis/normed_space/bounded_linear_maps.lean

```diff
@@ -22,15 +23,162 @@ local notation f ` →_{`:50 a `} `:0 b := filter.tendsto f (nhds a) (nhds b)
+  let ⟨M, hMp, hMb⟩ := has_pos_bound in ⟨M, hMp,
+  λ x, or.elim (lt_or_eq_of_le (norm_nonneg x))
+  (λ hlt, div_le_of_le_mul hlt (by rw mul_comm; exact hMb x))
+  (λ heq, by rw [←heq, div_zero]; exact le_of_lt hMp)⟩
+
+def of_linear_map_of_bounded {f : linear_map k E F}
+  (h : ∃ M, ∀ x, ∥f x∥ ≤ M * ∥x∥) : bounded_linear_map k E F := ⟨f, h⟩
+
+theorem is_bounded_linear_map {f: bounded_linear_map k E F} :
+  is_bounded_linear_map f :=
+  ⟨f.to_linear_map.is_linear,
+  let ⟨M, hMg, hMb⟩ := has_pos_bound in ⟨M, hMg, hMb⟩⟩
+
+-- the zero map and the identity maps are bounded
+def zero : bounded_linear_map k E F :=
+  ⟨0, 0, λ x, by rw zero_mul; exact le_of_eq norm_zero⟩
+
+def id : bounded_linear_map k E E :=
+  ⟨linear_map.id, 1, λ x, le_of_eq (one_mul _).symm⟩
+
+-- boundedness respects a bunch of operations
+def comp (g : bounded_linear_map k F G) (f : bounded_linear_map k E F) :
+  bounded_linear_map k E G :=
+  ⟨linear_map.comp g.to_linear_map f.to_linear_map,
+  let ⟨Mg, hMgp, hMgb⟩ := has_pos_bound in
+  let ⟨Mf, _, hMfb⟩ := has_pos_bound in ⟨Mg * Mf, λ x, by rw mul_assoc;
+  exact le_trans (hMgb _) ((mul_le_mul_left hMgp).2 (hMfb _))⟩⟩
+
+instance : has_zero (bounded_linear_map k E F) := ⟨zero⟩
+instance : has_one (bounded_linear_map k E E) := ⟨id⟩
+
+instance : has_mul (bounded_linear_map k E E) := ⟨comp⟩
+
+instance : has_add (bounded_linear_map k E F) :=
+  ⟨λ f g, ⟨f + g, let ⟨Mg, hMg⟩ := g.bounded in
+  let ⟨Mf, hMf⟩ := f.bounded in ⟨Mf + Mg, λ x,
+  calc _ ≤ ∥f x∥ + ∥g x∥       : norm_triangle _ _
+...      ≤ Mf * ∥x∥ + Mg * ∥x∥ : add_le_add (hMf _) (hMg _)
+...      = (Mf + Mg) * ∥x∥     : (add_mul _ _ _).symm ⟩⟩⟩
+
+instance : has_scalar k (bounded_linear_map k E F) :=
+  ⟨λ c f, ⟨c • f, let ⟨M, hM⟩ := f.bounded in ⟨∥c∥ * M,
+  λ x, by rw mul_assoc; exact (norm_smul c (f x)).symm ▸
+  (mul_le_mul_of_nonneg_left (hM x) (norm_nonneg c))⟩⟩⟩
+
+instance : has_neg (bounded_linear_map k E F) := ⟨λ f, (-1 : k) • f⟩
+instance : has_sub (bounded_linear_map k E F) := ⟨λ f g, f + (-g)⟩
+
+instance to_add_comm_group : add_comm_group (bounded_linear_map k E F) := {
+  add       := (+),
+  add_assoc := λ _ _ _, ext $ λ _, add_assoc _ _ _,
+  add_comm  := λ _ _, ext $ λ _, add_comm _ _,
+  zero      := 0,
+  add_zero  := λ _, ext $ λ _, add_zero _,
+  zero_add  := λ _, ext $ λ _, zero_add _,
+  neg       := λ f, -f,
+  add_left_neg := λ f, ext $ λ x, have t: (-1 : k) • f x + f x = 0, from
+    by rw neg_one_smul; exact add_left_neg _, t
+}
+
+instance : vector_space k (bounded_linear_map k E F) := {
+  smul_zero := λ _, ext $ λ _, smul_zero _,
+  zero_smul := λ _, ext $ λ _, semimodule.zero_smul _ _,
+  one_smul  := λ _, ext $ λ _, semimodule.one_smul _ _,
+  mul_smul  := λ _ _ _, ext $ λ _, semimodule.mul_smul _ _ _,
+  add_smul  := λ _ _ _, ext $ λ _, semimodule.add_smul _ _ _,
+  smul_add  := λ _ _ _, ext $ λ _, semimodule.smul_add _ _ _
+}
+
+variables {α : k} {u v : E}
+
+-- (helpful?) lemmas because invoking the corresponding results
```

Comment:
Indeed such lemmas are helpful. I think the comment doesn't need to end up in mathlib thought ;-)
```suggestion
```